### PR TITLE
AS-99 When UI is spanish, force the spanish to be first in priorities

### PR DESCRIPTION
**Pull request by YuriHechter**

### CODE REVIEW

File: search/engine.go [hunk 1]
```diff
@@ -494,14 +504,19 @@ func (e *ESEngine) DoSearch(ctx context.Context, query Query, sortBy string, fro
 				continue
 			}
 
+			highlightsLang := query.LanguageOrder
```
Comment (bbfsdev) [inline search/engine.go]: s/highlightsLang/highlightLangs/

File: api/handlers.go [hunk 2]
```diff
@@ -435,6 +435,17 @@ func SearchHandler(c *gin.Context) {
 		}
 	}
 
+	//  Quick workround to allow Spanish support when the interface language is Spanish (AS-99).
+	if c.Query("language") == consts.LANG_SPANISH {
```
Comment (bbfsdev) [inline api/handlers.go]: Why do you need this part?
You have hasSpanish below?

File: search/engine.go [hunk 3]
```diff
@@ -426,7 +426,13 @@ func (e *ESEngine) DoSearch(ctx context.Context, query Query, sortBy string, fro
 	if err != nil {
 		return nil, errors.Wrap(err, "ESEngine.DoSearch - Error multisearch Do.")
 	}
-
+	hasSpanish := false
```
Comment (bbfsdev) [inline search/engine.go]: Change this to shouldMergeResults and comment that we test this with Spanish for now.

File: search/engine.go [hunk 4]
```diff
@@ -473,9 +479,13 @@ func (e *ESEngine) DoSearch(ctx context.Context, query Query, sortBy string, fro
 	results := make([]*elastic.SearchResult, 0)
 	for _, lang := range query.LanguageOrder {
 		if r, ok := resultsByLang[lang]; ok {
-			results = r
-			currentLang = lang
-			break
+			if hasSpanish {
+				results = append(results, resultsByLang[lang]...)
```
Comment (bbfsdev) [inline search/engine.go]: You don't touch any scoring, right?
Which is ok I think.

Comment (bbfsdev) [reply search/engine.go]: Where are the different results being ordered by score?
Where that happens?

File: search/engine.go (hunk 1 above)
Comment (YuriHechter) [reply search/engine.go]: done

File: api/handlers.go (hunk 2 above)
Comment (YuriHechter) [reply api/handlers.go]: it is on a different file and package

File: search/engine.go (hunk 3 above)
Comment (YuriHechter) [reply search/engine.go]: done

File: search/engine.go (hunk 4 above)
Comment (YuriHechter) [reply search/engine.go]: Scores was not touched.
Ordering by score is inside joinResponses func. (using compareHits).

**Pull request closed without merging**